### PR TITLE
Add domains/tenants pattern matching feature to document

### PR DESCRIPTION
**Pull request by nodece**

### CODE REVIEW

File: docs/RBAC.md [hunk 1]
```diff
@@ -97,6 +97,20 @@ Casbin will automatically match ``/book/1`` and ``/book/2`` into pattern ``/book
...
+
+register `keyMatch2` to model:
+
+```
+m = g(r.sub, p.sub, r.dom) && keyMatch2(r.dom, p.dom) && r.obj == p.obj && r.act == p.act
+```
+
 You can see the full example [here](https://github.com/casbin/casbin/blob/dbdb6cbe2e7a80863e4951f9ff36da07fef01b75/model_test.go#L278-L307).
 
 :::note
```
Comment (hsluoyz) [inline docs/RBAC.md:116]: This note should be removed or revised.

File: docs/RBAC.md [hunk 2]
```diff
@@ -97,6 +97,20 @@ Casbin will automatically match ``/book/1`` and ``/book/2`` into pattern ``/book
 e.rm.(*defaultrolemanager.RoleManager).AddMatchingFunc("KeyMatch2", util.KeyMatch2)
 ```
 
+We also support the use of pattern matching function in domains/tenants, You need to register the function to enforcer and model.
```
Comment (hsluoyz) [inline docs/RBAC.md:100]: We should convey clear message that all 3 args can be wildcard.

**Comment by nodece** [comment]
@hsluoyz Please review again.

**Comment by hsluoyz** [comment]
lgtm

**Pull request merged**